fixtures/cloudflare-dev-module-resolution/index.test.ts

```diff
@@ -0,0 +1,50 @@
+import { createServer } from 'vite';
+import { beforeAll, describe, expect, test } from 'vitest';
+
+import config from './vite.config';
+
+describe('Cloudflare dev module resolution', () => {
+  let jsonOutput: Record<string, unknown> = {};
+
+  beforeAll(async () => {
+    jsonOutput = (await fetchJsonFromViteDevServer()) ?? {};
+  });
+
+  test('can successfully import "React"', () => {
+    expect(jsonOutput['typeof React']).toEqual('object');
+    expect(jsonOutput['typeof React.cloneElement']).toEqual('function');
+    expect(jsonOutput['reactVersionsMatch']).toEqual(true);
+  });
+
+  test('can successfully import utilities from "@remix-run/cloudflare"', () => {
+    expect(jsonOutput['type of remix cloudflare json({})']).toEqual('object');
+    expect(jsonOutput['remixRunCloudflareCookieName']).toEqual(
+      'my-remix-run-cloudflare-cookie',
+    );
+  });
+});
+
+async function fetchJsonFromViteDevServer(): Promise<null | Record<
+  string,
+  unknown
+>> {
```

Comment:
```suggestion
async function  fetchOutputFromViteDevServer(): Promise<null | unknown> {
```

Reply:
Sorry, seems I didn't save my suggestion here. I think the explicit return type should be removed (same reasons as in the tests):

```suggestion
async function fetchOutputFromViteDevServer() {
```

Reply:
I personally prefer the explicit return type (`Promise<null | unknown>`) here instead of adding `as unknown` on the return statement.

it feels a bit cleaner to me, but I'm happy to go with the `as unknown` if you strongly prefer that 🙂

Reply:
Having `Promise<unknown>` as the return type is fine.